### PR TITLE
Feature extract update

### LINKED ISSUE #23: Correct feature parsing
At the moment, we are extracting features based on the \w+\ regex query. This results in some non-ideal feature names. For example: Product names containing "latex-free" and "latexfree" are considered different and split into the "latex", "free" and "latexfree" features respectively.

### PR DESCRIPTION
**Pull request by Navbryce**
Changes:

- No longer extracting features from category and subcategory
- If the product name is formatted like a CSV, extract features like a "csv" -- each comma separated value is a feature
- "Singularize" plural feature names
- If an autocomplete query returns no result, try again with a singular version of the feature name. Justification: the user types a plural feature name too quickly before the singular autocomplete query comes up (most plural words contain singular word), so they accidentally filter out matching product feature results because they're now filtering by the plural version of the feature name

Unrelated small changes

- Remove papaparse as a dependency in package.json
- Fixed a small bug where responses weren't being returned with invalid parameters (had to do with an error being thrown outside a try catch, resulting in the server not returning a response)

Ex:
`localhost:3001/api/search/autocomplete?q=wrenches&features=` will try the query again with "wrench" instead

Closes #23

### CODE REVIEW

File: backend/resources/medshare-data-sample-reformat.csv [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ProductRef,ProductName,ProductCatID,Category Name,Sub Category,UnitMeasureName
-700001,Misc. Medical Supplies and Equipment,1,Medical Equipment,Unsorted,LB
+700001,"Wrench, Open End, 3 1/2""",1,Medical Equipment,Unsorted,LB
```
Comment (Arthelon) [inline backend/resources/medshare-data-sample-reformat.csv]: How did this change come about?

File: backend/src/utilities/feature-extractor.js [hunk 2]
```diff
@@ -93,25 +99,66 @@ async function processProductObjectAndInsertIntoDB(productObject) {
...
+  return extractFeaturesByRegex(normalizedValue, regex, removeList).map(normalizeFeatureAfterExtraction);
+}
+
+function removeSpecialCharactersAndGenerateOffsetList(value) {
+  const removeList = [{ index: -1, offset: 0 }];
+  let offset = 0;
+  let matchResult = REMOVE_REGEX.exec(value);
+  while (matchResult !== null) {
+    const removeIndex = matchResult.index;
+
```
Comment (Arthelon) [inline backend/src/utilities/feature-extractor.js]: I think one exception we should make here is for size ranges (i.e. 5-6). One way to tackle this is to check if chars both sides (excluding spaces) are integers and preserve the dash if that is the case. 

You could probably address this with Regex magic too.

File: backend/resources/medshare-data-sample-reformat.csv (hunk 1 above)
Comment (Navbryce) [reply backend/resources/medshare-data-sample-reformat.csv]: My bad. Was playing around with the data set

File: backend/src/utilities/feature-extractor.js (hunk 2 above)
Comment (Navbryce) [reply backend/src/utilities/feature-extractor.js]: Done with the ranges

**Pull request merged**